### ISSUE TITLE
Consult Mattermost roadmap for planning purposes

### ISSUE THREAD

**Issue by jtdaugherty** [matterhorn-chat/matterhorn#201]
For budgeting and planning purposes we need to account for upcoming mattermost releases. Our concerns are to maintain feature parity and compatibility. Releases in our next planning cycle are planned for May and July.

**Comment by jtdaugherty** [comment]
I'll be spending some time on 3/31 (4-5 pm) looking at the roadmap if anyone wants to join.

**Comment by jtdaugherty** [comment]
The result of this investigation was that the MatterMost roadmap itself has very little to say about the releases in our upcoming planning cycle. The longer-term roadmap has a list of features but at this point it's impossible to determine their impact on us. GitHub and Jira milestones for the releases planned in May and July also have very little to say about breaking changes or new features, so for now our plan is to be responsive to MatterMost changelog entries rather than plan against the upstream roadmap.

**Comment by jasonblais** [comment]
@jtdaugherty This sounds like an issue on our (Mattermost) end as we're not doing a great job communicating future releases and how they might impact people (users, developers and IT admins alike).

Wondering if your team would be interested to work together on how we can better communicate future roadmap items to our community, including yours?

As a start, we created a new webpage for [list of deprecated features, both past and upcoming](https://about.mattermost.com/deprecated-features/), which we'll start linking in our release announces, changelog and upgrade guide.

**Comment by jtdaugherty** [comment]
@jasonblais Thanks for that context! I hope it's clear that my comments on this ticket were more not intended as a criticism of the stuff I was looking at, but more just an observation that there didn't seem to be much that we needed to get ready for. But if what you're saying is true and there *is* more to the story for the upcoming releases, then yes, it would be helpful for that to be communicated! (Most of what I saw in the milestones I looked at was small UI tweaks and bug fixes, neither of which have much impact on our development.)

As for how to communicate going forward, I think listing upcoming features (with links to relevant tickets or milestones for details) on the roadmap page would be helpful. And if the ticket milestones in Jira or GitHub are incomplete, then certainly making sure that they include feature development efforts would also be helpful. The page you mentioned is password protected, so I can't look at it, unfortunately, but in the past we have benefitted from the detailed list of deprecated features in the changelog (*especially* details about upcoming deprecation plans).

On a related note, can you help us understand why there is a strong GitHub presence for MatterMost ticketing that occasionally has ties to a Jira presence? When reporting bugs or requesting features I find it 1) difficult to know where best to report a problem, and 2) frustrating when I report a problem on GitHub (where I would prefer to work) and then it is migrated to Jira. I would really prefer a single place to have those communications no matter what it is, though, so e.g. if Jira is really the "right" place to go, I'd prefer to see the MatterMost github issues disabled to avoid confusion.

**Comment by jasonblais** [comment]
Thanks @jtdaugherty for the feedback, sounds like the webpage for listing past and upcoming deprecated features is a good start. I've also removed the password protection, that was unintentional: https://about.mattermost.com/deprecated-features/

We'll continue using that page to update any known upcoming deprecations.

There has been a lot of internal discussion about our ticket system too -- we tried using GitHub for our [mattermost-mobile](https://github.com/mattermost/mattermost-mobile) project, but the GitHub kanban board proved to be insufficient for our needs. We use Jira because it allows easier prioritization and sprint planning.

Then why use GitHub? We primarily use it for help wanted issues as we noticed community members are more used to using GitHub than Jira.
